### DISCUSSION TITLE
Screen shot image resolution quality

### DISCUSSION THREAD

**Discussion by harlan3** [hizzgdev/jsmind#575] [General]
I have noticed that on large mind maps, when I take a snapshot using the shootDownload function in jsmind.screenshot.js, the file has poor image resolution and the text of the nodes is difficult to read. Is there some parameter that controls the image resolution quality? On smaller mindmaps the image resolution looks fine.

**Comment by hizzgdev** [comment]
sounds like a bug, will investigate it. BTW, there is not such an option.

**Reply by harlan3** [reply]
Thank you for looking into it.

**Reply by hizzgdev** [reply]
Hi @harlan3 , I can't reproduce the issue, could you share your test mindmap with me?
And, would you mind to try the case with the latest version of jsmind? this is the doc for screenshot https://github.com/hizzgdev/jsmind/blob/master/docs/en/plugin-screenshot.md.

**Reply by harlan3** [reply]
See attached zip file for my example freemind mindmap. I generated the freemind mindmap from a Python script. I use this mindmap with an an older 0.3 version of jsmind which has been modified to support clicking on node links, to pull up a book or magazine from my library. For the latest version of jsmind, I am trying to load it from jsmind/example/2_features.html section MultiFormat C - freemind. However, with this latest version of jsmind, the lines connecting the nodes are not being drawn so I can't really test the screenshot capability. Everything works properly on my older version 0.3 of jsmind. Maybe there are too many nodes? I remember having a similar problem before, and if I remember correctly, it required switching from the default javascript canvas to SVG mode.

[Freemind_Content.zip](https://github.com/hizzgdev/jsmind/files/14337533/Freemind_Content.zip)

**Reply by harlan3** [reply]
Another observation, when I load the example freemind mindmap from the attachment above, it doesn't have any of the connecting lines between nodes. However, if I then collapse all nodes and zoom in or out, then the lines are then drawn properly. So I think jsmind is overloaded when all nodes are expanded by default. When I try to do a Screenshot after getting the node lines to appear, the javascript hangs on the page, which again seems like it is overloaded.

**Reply by hizzgdev** [reply]
Hi Harlan, It's really a huge mindmap, you have to use `SVG` mode to render it, because web browser can not handle such a large `canvas`, and unfortunately jsmind screenshot depends on `canvas` also. So It's not surprising that there are problems taking screenshot for this mind map.
I will investigate how to handle big mindmaps better, but it's not a short-term work.

**Reply by hizzgdev** [reply]
for your use case, you can leverage the capability of Google Chrome, see below
<img width="331" alt="image" src="https://github.com/hizzgdev/jsmind/assets/1690290/90785290-e0f8-4be2-ae68-bbbe6c35d97a">

You may need to try more times as my browser crashed when I was trying. Good luck!

**Reply by harlan3** [reply]
Thanks for this info on using Google Chrome for the screenshot. That seems to work pretty well.

**Reply by harlan3** [reply]
I think there are a couple of reasons why it was working better for me on the older jsmind v0.3. Like you mentioned, I had modified the older version to use the SVG renderer instead of Javascript canvas. The other reason for better performance is that the latest version of jsmind does not seem to acknowledge the node attribute FOLDED="true" for all of the nodes in the mindmap and instead tries to start with all nodes expanded by default. Not sure if the ignoring of FOLDED attribute is a bug?

**Reply by hizzgdev** [reply]
Great thanks @harlan3, the `folded` is just ignored, I need to recheck the parsing logic of the freemind file.

**Reply by hizzgdev** [reply]
`folded` is supported in https://github.com/hizzgdev/jsmind/pull/577
